src/main/java/in/projecteka/consentmanager/user/SignUpRequestValidator.java

```diff
@@ -37,16 +43,48 @@ private SignUpRequestValidator() {
                 validate(signUpRequest.getGender()),
                 validateUserName(signUpRequest.getUsername(), userIdSuffix),
                 validatePassword(signUpRequest.getPassword()),
-                validateYearOfBirth(signUpRequest.getYearOfBirth()))
-                .ap((firstName, gender, username, password, dateOfBirth) -> SignUpRequest.builder()
+                validateYearOfBirth(signUpRequest.getYearOfBirth()),
+                validateUnVerifiedIdentifiers(signUpRequest.getUnverifiedIdentifiers()))
+                .ap((firstName, gender, username, password, dateOfBirth, unverifiedIdentifiers) -> SignUpRequest.builder()
                         .name(firstName)
                         .gender(gender)
                         .username(username)
                         .password(password)
                         .yearOfBirth(dateOfBirth)
+                        .unverifiedIdentifiers(unverifiedIdentifiers)
                         .build());
     }
 
+    protected static Validation<String,List<Identifier>> validateUnVerifiedIdentifiers(List<Identifier> unverifiedIdentifiers) {
+        if (unverifiedIdentifiers==null || unverifiedIdentifiers.isEmpty()) {
+            return Validation.valid(unverifiedIdentifiers);
+        }
+
+        List<Identifier> abIdentifiers = unverifiedIdentifiers.stream().
+                filter(identifier -> identifier.getType().equals(IdentifierType.ABPMJAYID)).
+                collect(Collectors.toList());
```

Comment:
can you export this as a separate method isUnique(identifierType), where we can pass identifier type and list, and returns true or false?

Reply:
Done.